dd-sdk-android-ndk/src/main/cpp/utils/signal-monitor.c

```diff
@@ -194,28 +192,25 @@ bool init_datadog_signal_handler() {
     if (datadog_sigaction == NULL) {
         return false;
     }
-    sigemptyset((sigset_t *) &datadog_sigaction->sa_mask);
+    if (sigemptyset((sigset_t *) &datadog_sigaction->sa_mask) != 0) {
+        return false;
+    }
     datadog_sigaction->sa_sigaction = handle_signal;
     // we will use the SA_ONSTACK mask here to handle the signal in a freshly new stack.
     datadog_sigaction->sa_flags = SA_SIGINFO | SA_ONSTACK;
 
     return true;
 }
 
-/*
- * Overriding the signal handlers from the new spawned thread. This will allow us to block (handle)
- * also the `SIQGUIT` signal triggered when the process is unresponsive - ANR.
- * This function should be called inside a locked mutex for Thread safety
- */
 bool override_native_signal_handlers() {
-    if (handlers_installed) {
+    if (!init_datadog_signal_handler()) {
+        __android_log_write(ANDROID_LOG_ERROR, LOG_TAG, SIGNAL_HANDLERS_INIT_ERROR_MESSAGE);
```

Comment:
There are multiple way `init_datadog_signal_handler`, it'd be nice to have more verbose error message specific to what went wrong.